### PR TITLE
Describe the obstacles options of labels

### LINKED ISSUE #617: [labeling] Add option to control how polygon layers     act as obstacles
https://github.com/qgis/QGIS/commit/3a44e294de08ee03c0386179693c61a3f76e5efe
Author: Nyall Dawson
Date:   Fri Jul 17 15:26:57 2015 +1000
    
    Options are either avoid placing labels over polygon interior
    or avoid placing over polygon boundaries. (Previous behaviour
    was always avoid placing over interior).
    
    Avoiding placing over boundaries is useful for regional boundary
    layers, where the features cover an entire area. In this case
    it's impossible to avoid placing labels within these features,
    and it looks much better to avoid placing them over the boundaries
    between features. End result is better cartographic placement of
    labels in this situation.

### LINKED ISSUE #619: [labeling] Data defined control over whether a feature acts as an obstacle for labels
https://github.com/qgis/QGIS/commit/c97733ea27e6db1f1278e439a0d52ed33a9478a5
Author: Nyall Dawson
Date:   Thu Jul 23 15:00:45 2015 +1000

### LINKED ISSUE #620: [labeling] Add priority control for obstacles
https://github.com/qgis/QGIS/commit/b4311877f72bdb6516080011c7aa0c3c0d108307
Author: Nyall Dawson
Date:   Thu Jul 23 15:00:53 2015 +1000

    Allows you to make labels prefer to overlap features from certain
    layers rather than others. Can also be data defined, so that certain
    features are more likely to be covered than others.

### PR DESCRIPTION
**Pull request by DelazJ**
fix #617, fix #619, fix #620 

@nyalldawson can you check this and let me know if it covers and explains as expected how you have designed this frame, please?
Thanks

### CODE REVIEW

**Comment by nyalldawson** [comment]
Looks good to me!

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 1]
```diff
@@ -1259,9 +1259,29 @@ to avoid duplicate labels`, rendering a quite airy map in conjunction with
 the :guilabel:`Distance` or :guilabel:`Repeat` options in Placement tab.
 
 
-Under :guilabel:`Obstacles`, the |checkbox| :guilabel:`Discourage
-labels from covering features` helps you manage covering relation
-between labels and other features.
+From the :guilabel:`Obstacles` frame, you can manage the covering relation between
+labels and features. Activate the |checkbox| :guilabel:`Discourage
+labels from covering features` option to decide whether features of the layer
+should act as obstacles for labels or not. Selection of features to use as obstacle
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: Another question, if I may: when features of a layer are set as obstacles, is it only against labels of other layers or against any label (including their own ones). I mean, do the placement of that layer labels take into account what is set as obstacle parameters in the layer properties or just done according to other options of placement?
I don't know if I'm clear :(, but e.g., is it `acts as obstacles for labels` or `acts as obstacles for labels of other layers`

Comment (nyalldawson) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: It's an obstacle for any labels (ie including current layer labels) - but NOT for the feature being labeled. Eg a feature won't be an obstacle for its own label.

Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: Ok, clear.
Thanks.

**Comment by yjacolin** [comment]
@DelazJ are you ok with this PR? Can you merge it?

**Comment by DelazJ** [comment]
@yjacolin No I wasn't :). This feature is one I wasn't sure to understand the behavior till I wrote this PR. So I reword and add some precision. HTH.

**Pull request merged**